src/cpu/dyn_cache.h

```diff
@@ -741,16 +792,26 @@ static void cache_init(bool enable) {
 		if (cache_code_start_ptr==NULL) {
 			// allocate the code cache memory
 #if defined (WIN32)
-			cache_code_start_ptr=(Bit8u*)VirtualAlloc(0,CACHE_TOTAL+CACHE_MAXSIZE+PAGESIZE_TEMP-1+PAGESIZE_TEMP,
+			cache_code_start_ptr=(Bit8u*)VirtualAlloc(0,cache_code_start_sz,
 				MEM_COMMIT,PAGE_EXECUTE_READWRITE);
 			if (!cache_code_start_ptr)
-				cache_code_start_ptr=(Bit8u*)malloc(CACHE_TOTAL+CACHE_MAXSIZE+PAGESIZE_TEMP-1+PAGESIZE_TEMP);
+				cache_code_start_ptr=(Bit8u*)malloc(cache_code_start_sz);
 #else
-			cache_code_start_ptr=(Bit8u*)malloc(CACHE_TOTAL+CACHE_MAXSIZE+PAGESIZE_TEMP-1+PAGESIZE_TEMP);
+			int map_flags = MAP_PRIVATE | MAP_ANON;
+			int prot_flags = PROT_READ | PROT_WRITE | PROT_EXEC;
+#if defined(HAVE_MAPJIT)
+			map_flags |= MAP_JIT;
 #endif
+#if defined(HAVE_MMAP)
+			cache_code_start_ptr=(Bit8u*)mmap(NULL, cache_code_start_sz, prot_flags, map_flags, -1, 0);
```

Comment:
Prefer `static_cast<uint8_t*>(..)` and `nullptr` instead of NULL (justification  --  https://www.codespeedy.com/difference-between-null-and-nullptr-in-cpp/).

Reply:
this was the original code pre-patch, so that raises a good coding standard question - upgrade to newer constructs as you edit "nearby" code, leaving a patchwork throughout a file, or take the time to upgrade the entire file?

Reply:
We use git SVN's integration to apply commits from upstream - so the deeper the edits on our side, the more manual futzing is needed to get those applied. 
(So that's why we haven't done search and replaces or mass reformatting). 
For big files like this and a PR with laser purpose intent, the patchwork approach is best.